core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -147,6 +147,14 @@ default void finishTableExecute(ConnectorSession session, ConnectorTableExecuteH
         throw new TrinoException(GENERIC_INTERNAL_ERROR, "ConnectorMetadata getTableHandleForExecute() is implemented without finishTableExecute()");
     }
 
+    /**
+     * Execute table execute on coordinator node only
+     */
+    default void executeTableExecute(ConnectorSession session, ConnectorTableExecuteHandle tableExecuteHandle)
+    {
+        throw new TrinoException(GENERIC_INTERNAL_ERROR, "ConnectorMetadata getTableHandleForExecute() is implemented without executeTableExecute()");
```

Comment:
It's not true that implementing `getTableHandleForExecute` implies need to implement this method. Change message wording.